### ISSUE TITLE
Question about training, validation, testing and missing rates?

### ISSUE THREAD

**Issue by Rajesh90123** [WenjieDu/SAITS#37]
Greetings, sir.
In your research paper, "SAITS: Self-Attention-based Imputation for Time Series", you have presented a table with different missing rates (20% to 90%). So, do you prepare best models from each val_dataset with different missing % rates so as to provide result for corresponding missing % of test or do you save model based on specific missing percentage in validation data (say 20% loss in val data) and use it to prepare result for test for other missing percentage(say 30% artificial missing percentage in test data)?

**Comment by WenjieDu** [comment]
Hi there,<br><br>Thank you so much for your attention to SAITS! If you find SAITS is helpful to your work, please star⭐️ this repository. Your star is your recognition, which can let others notice SAITS. It matters and is definitely a kind of contribution.<br><br>I have received your message and will respond ASAP. Thank you again for your patience! 😃<br><br>Best,<br>Wenjie

**Comment by WenjieDu** [comment]
You can figure this out by reading code in our data preprocessing scripts under the dir [dataset_generating_scripts](https://github.com/WenjieDu/SAITS/tree/main/dataset_generating_scripts)

**Comment by Rajesh90123** [comment]
Thank you sir for your response. And do you also find best hyperparameters for each % missing rate of validation set to test for each % missing rate in test dataset?

**Comment by WenjieDu** [comment]
Yep, this is also clarified in our paper. I suggest you use PyPOTS https://github.com/WenjieDu/PyPOTS, which could be quite useful to you if you work on incomplete time series with missingness.
